scripts/manpage_generation.sh

```diff
@@ -5,15 +5,41 @@ MANPAGE_DIR="man"
 
 echo Generating manpages
 
-VER=$(grep -oP '^Version:\s+\K\S+' packaging/convert2rhel.spec)
+# Detect changes in __init__.py
+if git diff --quiet HEAD~1 -- convert2rhel/__init__.py; then
+    # No changes detected, exit with 0
+    echo "No new version detected"
+    exit 0
+else
+    echo "enter changes deteced"
+    # Changes detected, generate manpages
 
-echo Generating for version $VER
-# Generate a file with convert2rhel synopsis for argparse-manpage
-/usr/bin/python -c 'from convert2rhel import toolopts; print("[synopsis]\n."+toolopts.CLI.usage())' > man/synopsis
+    #get dependcy
+    import subprocess
 
-/usr/bin/python -m pip install argparse-manpage six pexpect
+    # Install Python packages
+    subprocess.run(["pip", "install", "argparse-manpage", "six", "pexpect"])
```

Comment:
This can be moved out to a `requirements.txt` file, can get inspiration from other requirement files in the repo. To get it from the container directly you can do `pip freeze -f requirements.txt`. Then add this as a step in the container, this way we can cache it

Reply:
Action is also failing here due to syntax error

https://github.com/oamg/convert2rhel/actions/runs/9072412663/job/24927859923

Reply:
Gotcha I can get that moved out to an requirements.txt file! I will also get the action fixed too!

Thank you for the help with theses changes!